### PR TITLE
Add integration tests for user-registrants retrieval endpoint

### PR DESCRIPTION
**Pull request by pamodaaw**
This PR adds integration tests for the user-registrants retrieval endpoint in the application management API. 

The API layer changes were introduced with https://github.com/wso2/identity-api-server/pull/417
The BE improvements were introduced with https://github.com/wso2-extensions/identity-governance/pull/663

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("totalResults", equalTo(0))
+                .body("userRegistrants.size()", equalTo(0));
+    }
+
+    @Test
+    public void testCreateAppWithNoAuthenticationOptions() throws Exception {
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java:78]: Shall we do this also like doPutAuthenticatorsAndVerify. Is there a need to add this as a test?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    public void testRetrieveUserRegistrantsDefault() {
+
+        Response response = getResponseOfGet(APPLICATION_MANAGEMENT_API_BASE_PATH + PATH_SEPARATOR + createdAppId +
+                APPLICATION_USER_REGISTRANT_CONTEXT);
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("totalResults", equalTo(1))
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java:104]: Wondering whether we should check the default is properly coming. We currently have a fixed default so checking that should be okay.

Comment (pamodaaw) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java:104]: Did you mean that always the default will be the Basic Authenticator?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java (hunk 1 above)
Comment (pamodaaw) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java:78]: Is there any disadvantage in having it as a test?

Comment (janakamarasena) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java:80]: No disadvantage. That particular test is not relevant to this context

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java (hunk 2 above)
Comment (janakamarasena) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/application/management/v1/ApplicationManagementUserRegistrantSuccessTest.java]: Yes always basic authenticator will be the default according to current configurations.

**Review by janakamarasena — APPROVED** [review]

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**